backend/src/main/java/com/jobwait/jackson/AnswerDeserializer.java

```diff
@@ -0,0 +1,60 @@
+package com.jobwait.jackson;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.jobwait.domain.Answer;
+import com.jobwait.domain.Questions;
+import com.jobwait.spring.utils.Utils;
+import com.jobwait.domain.AnswerType;
+import com.jobwait.domain.Question;
+
+public class AnswerDeserializer extends StdDeserializer<Answer> {
+    public AnswerDeserializer() {
+        this(null);
+    }
+
+    public AnswerDeserializer(Class<?> vc) {
+        super(vc);
+    }
+
+    @Override
+    public Answer deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException, JacksonException {
+        JsonNode node = jp.getCodec().readTree(jp);
+
+        JsonNode questionKeyNode = node.get("questionKey");
+        if (questionKeyNode == null) {
+            throw new IOException("questionKey not found");
```

Comment:
i think there is a JsonProcessingException you can use here instead that would make it more readable/clear, it should be a part of Jackson

Reply:
I ran into problems trying to use that exception, I'll try again.
I agree, IOException is a terrible choice.